x-pack/test/functional/services/ml/job_selection.ts

```diff
@@ -23,5 +23,18 @@ export function MachineLearningJobSelectionProvider({ getService }: FtrProviderC
         `Job selection should display jobs or groups '${jobOrGroupIds}' (got '${actualJobOrGroupLabels}')`
       );
     },
+
+    async assertJobSelectionNotExist(jobOrGroupIds: string[]) {
+      const selectedJobsOrGroups = await testSubjects.findAll(
+        'mlJobSelectionBadges > ~mlJobSelectionBadge'
+      );
+      const actualJobOrGroupLabels = await Promise.all(
+        selectedJobsOrGroups.map(async (badge) => await badge.getVisibleText())
+      );
+      expect(actualJobOrGroupLabels).to.not.eql(
```

Comment:
The current validation would still be successful for the following example:
- assertJobSelectionNotExist(['jobIdA'])
- actualJobOrGroupLabels = ['jobIdA', 'jobIdB']

I think in most cases we want to know that a specific jobId is not part of the current job selection. So my suggestion would be to rename this method to something like `assertJobSelectionNotContains(jobGroupOrId: string)` and then validate `expect(actualJobOrGroupLabels).to.not.contain(jobGroupOrId, '<message>')`.

Reply:
Updated here https://github.com/elastic/kibana/pull/84594/commits/49899e9fda80eb3da31d70bef612eb9401346b91